solc/CommandLineInterface.cpp

```diff
@@ -1097,6 +1097,21 @@ void CommandLineInterface::outputCompilationResults()
 {
 	solAssert(m_options.input.mode == InputMode::Compiler || m_options.input.mode == InputMode::CompilerWithASTImport, "");
 
+	std::set<std::string> uncompiledContracts;
+	for (auto const& contract: m_compiler->contractNames())
+		if (objectWithLinkRefsHex(m_compiler->object(contract)).empty())
+			uncompiledContracts.insert(contract.substr(contract.find_last_of('/') + 1));
+
+	if (uncompiledContracts.size() == m_compiler->contractNames().size())
+		sout() << "Input did not contain any contracts to compile." << endl;
+	else
+		for (auto const& uncompiledContract: uncompiledContracts)
```

Comment:
The type is very short and readable here so no point in hiding it behind `auto`:
```suggestion
	for (string const& contract: m_compiler->contractNames())
		if (objectWithLinkRefsHex(m_compiler->object(contract)).empty())
			uncompiledContracts.insert(contract.substr(contract.find_last_of('/') + 1));

	if (uncompiledContracts.size() == m_compiler->contractNames().size())
		sout() << "Input did not contain any contracts to compile." << endl;
	else
		for (string const& uncompiledContract: uncompiledContracts)
```